pygmt/x2sys.py

```diff
@@ -14,10 +16,30 @@
     dummy_context,
     fmt_docstring,
     kwargs_to_strings,
+    unique_name,
     use_alias,
 )
 
 
+@contextlib.contextmanager
+def tempfile_from_dftrack(track, suffix):
+    """
+    Saves a pandas.DataFrame track table to a temporary tab-separated ASCII
+    text file with a suffix (e.g. 'xyz').
+    """
+    try:
+        tmpfilename = f"track-{unique_name()[:7]}.{suffix}"
+        track.to_csv(
+            path_or_buf=tmpfilename,
+            sep="\t",
+            index=False,
+            date_format="%Y-%m-%dT%H:%M:%S.%fZ",
+        )
+        yield tmpfilename
+    finally:
+        os.remove(tmpfilename)
```

Comment:
The original implementation using GMTTempFile/NamedTemporaryFile didn't work because of some permissions issues (on macOS/Windows), which is why this try-finally block is used.

Reply:
The code quality looks good. As you're the one who develops and uses these functions, we have to trust you. 😄 

Just one suggestion, add the comment to the codes, explaining why you use `unique_name` here. 

That's the first question when I read your codes before I see your comment here.

Reply:
> The code quality looks good. As you're the one who develops and uses these functions, we have to trust you. smile

It's all Paul's work done a decade ago, I'm just wrapping it in Python so more people can use it easily :smiley: You won't believe how many 'crossover analysis' tools have been written again and again, but that's another story.

> Just one suggestion, add the comment to the codes, explaining why you use `unique_name` here.
> 
> That's the first question when I read your codes before I see your comment here.

Ok, will do.